### ISSUE TITLE
Provide entry titles in the app to make entries more human readable

### ISSUE THREAD

**Issue by sandysage** [mozilla-lockbox/lockbox-ios#193]
When viewing the entry list, I want to quickly scan the entry list by title (e.g., Facebook instead of https://www.facebook.com or even www.facebook.com), so I can quickly access my username and password for logging in. 

Problem: No Entry Name (title) currently available in Saved Logins so we need to derive one

## Acceptance criteria
* Should be able to view the list of entries by the origin
* Default alphabetical ordering on the origin should make sense (eg facebook comes before twitter)
* Should not not include the protocol (http://, https://, www.)
* Should not see the suffix or anything after the suffix (.com/{*})
* Should support future goals of editing the entry name #174

## Potential solutions
1. Solution A: funky string manipulation, get domain name, drop suffix, close enough? www.chase.com = Chase, www.bankofamerica.com = Bankofamerica (Medium, not dependent)
2. Solution B: actually change the schema to include entry title (High, very dependent, long term roadmap)
3. Solution C: drop off protocol (drop https://) just show origin, just show TLD? Examples: chase.com, bankofamerica.com, but google.com could be accounts.google.com, mail.google.com (Low)

**Comment by devinreams** [comment]
@ssage why is this "Blocked"? Are you wanting to provide more context or input?

Or aren you looking for more input from eng into the potential solutions? If the latter, I think we should just put this into the backlog to begin exploring.

**Comment by sandysage** [comment]
@devinreams I had it as blocked since this needs input on the potential solutions - ie it's not ready to actually implement. It's now in the backlog!

**Comment by nickbrandt** [comment]
Solution A is the desired approach for MVP (from design perspective)

**Comment by devinreams** [comment]
@linuxwolf to drop in notes here...

**Comment by devinreams** [comment]
Thanks @linuxwolf 

@mozilla-lockbox/product given the input above, do you still prefer us going towards Solution A (versus C) for MVP?

**Comment by devinreams** [comment]
I quickly grabbed some data so we made sure we were talking about satisfying as many real world cases (domains) as possible and not over-optimizing around edge cases right now. Remembering the goal here is a simple solution to show a more legible (to humans) "title" than just a full domain, path, port, etc. derived from that data itself.

Source: [Top 500 sites according to Cisco Umbrella](http://s3-us-west-1.amazonaws.com/umbrella-static/index.html) (similar to Alexa 500) 

https://docs.google.com/spreadsheets/d/1iN_l__5VdTxQoMVQEjaAcxxnmau7LzKfiol2kkwZxLg/edit#gid=0


## Suffixes

The only suffixes present in the top 500 are:

- co
- com
- fi
- goog
- io
- net
- org
- tv

To @linuxwolf's point, which is spot on: we should acknowledge the need to update the suffixes, but it doesn't seem like it needs to be maintained very frequently.

In other words: we can assume what is published today should catch the majority of sites people are saving passwords for and _semi-infrequent updates_ (even if manual) would be acceptable, right?

## Solution A

You'll see this looks pretty good but there's a lot of duplication just because the big players own a lot of properties with the same top level domain (eg: google.com is repeated 29 times).


## Solution C

Similar problem, there are less total duplications but giving one more signal (suffix) may be useful. Adding a sub-domain may be even more helpful to consider here (we could consider a Solution C+).

---

I hope this is useful for @mozilla-lockbox/product to help decide on how to proceed.
